executor/analyze.go

```diff
@@ -691,7 +741,7 @@ func (e *AnalyzeFastExec) buildSampTask() (needRebuild bool, err error) {
 		targetKey = loc.EndKey
 
 		// If the KV pairs in the region all belonging to the table, add it to the sample task.
-		if bytes.Compare(startKey, loc.StartKey) <= 0 && bytes.Compare(loc.EndKey, endKey) <= 0 {
+		if bytes.Compare(startKey, loc.StartKey) <= 0 && len(loc.EndKey) != 0 && bytes.Compare(loc.EndKey, endKey) <= 0 {
```

Comment:
When will `loc.EndKey` be empty?

Reply:
If the end key of a region is empty, it represents infinity large.